lib/foreman_tasks_core/otp_manager.rb

```diff
@@ -9,18 +9,19 @@ def generate_otp(username)
         passwords[username] = otp.to_s
       end
 
-      def drop_otp(username, password)
-        passwords.delete(username) if passwords[username] == password
-      end
-
       def passwords
         @password ||= {}
       end
 
-      def authenticate(hash)
+      def authenticate(hash, expected_user: nil, clear: true)
         plain = Base64.decode64(hash)
         username, otp = plain.split(':', 2)
-        drop_otp(username, otp)
+        if expected_user
+          return false unless expected_user == username
+        end
```

Comment:
Any reason why not to combine these two conditions into one?

Reply:
Reading some golang code recently :) but I like the two conditions - one checking if we actually care about the equality, and the second for the matching itself. So I prefer to keep it, but I can compress it to one if that's too much for you :)

Reply:
No problem with it if was intentional. It just felt like there were originally more things in the outer condition and this is a leftover.